### ISSUE TITLE
Successive associations issue when migrating to rails 6.1

### ISSUE THREAD

**Issue by Paul-Yves** [thoughtbot/factory_bot_rails#391]
### Description

We have the following model relation :
```ruby
FactoryBot.define do
  factory :response_set_section_fieldset, class: 'ResponseSetSectionFieldset' do
    association :response_set_section, factory: :response_set_section, strategy: :build
  end
end

FactoryBot.define do
  factory :response_set_section, class: 'ResponseSetSection' do
    association :response_set, factory: :response_set, strategy: :build
  end
end
```
When building a response_set_section_fieldset, response_set_section_fieldset.response_set_section.response_set is created in rails 6.0 but upgrading to rails 6.1 result in response_set_section_fieldset.response_set_section.response_set being nil. Switching to create strategy prevent being nil but is not desirable due to long execution time.

Also note that if I build a response_set_section, response_set_section.response_set is not nil.

### Reproduction Steps

See above, I will try to isolate the behaviour in a small project if I have time.

### Expected behavior

response_set_section_fieldset.response_set_section.response_set should exist (chained association should work)

### Actual behavior

response_set_section_fieldset.response_set_section.response_set is nil whereas response_set_section.response_set is not nil (chained association seems to be broken)

### System configuration
**factory_bot_rails version**:  6.1.0
**factory_bot version**:  6.1.0
**rails version**:  from 6.0.3.5 to 6.1.2.1
**ruby version**: 2.7.2

**Comment by composerinteralia** [comment]
Nothing comes to mind looking at that code snippet. Based on the fact that this came up during a Rails upgrade, it seems likely this was caused by a change in Rails rather than a bug in factory_bot.

I'm going to close this for now, but can reopen and help debug if you are able to provide a reproduction script. Without that, there's not much I can do to help.

You could also try out building the objects without factory_bot to see if you get the same behavior. Given your code snippet above,

```rb
FactoryBot.create(:response_set_section_fieldset)
```

factory_bot is basically doing:

```rb
response_set = ResponseSet.new
# Assign attributes one by one, if the response set factory has any
response_set_section = ResponseSetSection.new
response_set_section.response_set = response_set
response_set_section_fieldset = ResponseSetSectionFieldset
response_set_section_fieldset.response_set_section = response_set_section
```

Thanks!

**Comment by Paul-Yves** [comment]
Building the objetcs without factory_bot does not produce the same error, I will try to identify what goes wrong exactly and maybe provide an isolated example

**Comment by Paul-Yves** [comment]
Ok I identified my issue. The tests were creating other objects with relationship to response_set_section_fieldset, using create strategy rather than build. Turns out it was mixing badly, something like 
```ruby
FactoryBot.define do
  factory :response, class: 'Response' do
    association :response_set_section_fieldset, factory: :response_set_section_fieldset, strategy: :build
  end
end
```
instanciated through create will save the response_set_section_fieldset but not the response_set_section (maybe something was invalid along the way). Using build everywhere solves the issue.
